internal/runners/pull/pull.go

```diff
@@ -212,6 +214,45 @@ func (p *Pull) performMerge(strategies *mono_models.MergeStrategies, remoteCommi
 	return resultCommit, nil
 }
 
+// mergeBuildScript merges the local build script with the remote buildexpression (not script) for a
+// given UUID, performing the given merge strategy (e.g. from model.MergeCommit).
+func (p *Pull) mergeBuildScript(strategies *mono_models.MergeStrategies, remoteCommit strfmt.UUID, localCommit strfmt.UUID) error {
+	// Verify we have a build script to merge.
+	script, err := buildscript.NewScriptFromProjectDir(p.project.Dir())
+	if err != nil && !buildscript.IsDoesNotExistError(err) {
+		return errs.Wrap(err, "Could not get local build script")
+	}
+	if script == nil {
+		return nil // no build script to merge
+	}
+
+	// Get the local and remote build expressions to merge.
+	bp := model.NewBuildPlannerModel(p.auth)
+	exprA, err := bp.GetBuildExpression(p.project.Owner(), p.project.Name(), localCommit.String())
+	if err != nil {
+		return errs.Wrap(err, "Unable to get buildexpression for local commit")
+	}
+	exprB, err := bp.GetBuildExpression(p.project.Owner(), p.project.Name(), remoteCommit.String())
+	if err != nil {
+		return errs.Wrap(err, "Unable to get buildexpression for remote commit")
+	}
+
+	// Attempt the merge.
+	mergedExpr, err := merge.Merge(exprA, exprB, strategies)
+	if err != nil {
+		err := buildscriptRunbits.GenerateAndWriteDiff(p.project, script, exprB)
+		if err != nil {
+			return locale.WrapError(err, "err_diff_build_script", "Unable to generate differences between local and remote build script")
+		}
+		return locale.NewInputError(
+			"err_build_script_merge",
+			"Unable to automatically merge build scripts. Please resolve conflicts manually and then run [ACTIONABLE]`state commit`[/RESET]")
```

Comment:
Please specify what file they need to resolve the conflicts in. I would make a code suggestion but I don't actually remember what we called it for the interim.